pinot-common/src/main/java/org/apache/pinot/common/utils/PinotDataType.java

```diff
@@ -538,6 +540,64 @@ public String convert(Object value, PinotDataType sourceType) {
     }
   },
 
+  JSON {
+    @Override
+    public int toInt(Object value) {
+      return Integer.parseInt(value.toString().trim());
+    }
+
+    @Override
+    public long toLong(Object value) {
+      return Long.parseLong(value.toString().trim());
+    }
+
+    @Override
+    public float toFloat(Object value) {
+      return Float.parseFloat(value.toString().trim());
+    }
+
+    @Override
+    public double toDouble(Object value) {
+      return Double.parseDouble(value.toString().trim());
+    }
+
+    @Override
+    public boolean toBoolean(Object value) {
+      return Boolean.parseBoolean(value.toString().trim());
+    }
+
+    @Override
+    public Timestamp toTimestamp(Object value) {
+      return (value instanceof Long) ? new Timestamp(((Long) value).longValue())
+          : Timestamp.valueOf(value.toString().trim());
+    }
+
+    @Override
+    public String toString(Object value) {
+      return value.toString();
+    }
+
+    @Override
+    public byte[] toBytes(Object value) {
+      if (value instanceof String) {
```

Comment:
Use `value.toString()` here to be consistent with other methods? Value should always be of string type here

Reply:
Fixed.